datadog/resource_datadog_webhook.go

```diff
@@ -0,0 +1,204 @@
+package datadog
+
+import (
+	"context"
+	"io/ioutil"
+	"strings"
+
+	"github.com/terraform-providers/terraform-provider-datadog/datadog/internal/utils"
+	"github.com/terraform-providers/terraform-provider-datadog/datadog/internal/validators"
+
+	datadogV1 "github.com/DataDog/datadog-api-client-go/api/v1/datadog"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func resourceDatadogWebhook() *schema.Resource {
+	return &schema.Resource{
+		Description:   "Provides a Datadog webhook resource. This can be used to create and manage Datadog webhooks.",
+		CreateContext: resourceDatadogWebhookCreate,
+		ReadContext:   resourceDatadogWebhookRead,
+		UpdateContext: resourceDatadogWebhookUpdate,
+		DeleteContext: resourceDatadogWebhookDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Description: "The name of the webhook. It corresponds with `<WEBHOOK_NAME>`.",
+				Type:        schema.TypeString,
+				Required:    true,
+			},
+			"url": {
+				Description: "The URL of the webhook.",
+				Type:        schema.TypeString,
+				Required:    true,
+			},
+			"payload": {
+				Description: "The payload of the webhook.",
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+			},
+			"custom_headers": {
+				Description: "The headers attached to the webhook.",
+				Type:        schema.TypeString,
+				Optional:    true,
+			},
+			"encode_as": {
+				Description:      "Encoding type. Can be given either `json` or `form`.",
```

Comment:
```suggestion
				Description:      "Encoding type.",
```
Allowed enum values automatically described when generating docs thanks to the validateFunc